kmod/src/data.c

```diff
@@ -1889,6 +1889,127 @@ int scoutfs_data_waiting(struct super_block *sb, u64 ino, u64 iblock,
 	return ret;
 }
 
+/* Mostly cribbed from mm/filemap.c:filemap_page_mkwrite() */
```

Comment:
Is what's cribbed still appropriate for the modern kernel paths?  Checking that would probably resolve the later patches that seem uncertain about whether or not they're doing it right.  See where the current vfs and file systems do the things this cribbed code, and those fixes, are doing and make ours do the same.

Reply:
I removed this comment entirely.

In a way this comment makes sense since filemap_page_mkwrite() literally is a skeleton function here but there's not a single filesystem that actually uses it, it's only used by `aio_ring`.

It is however still appropriate because it shows the basic locking order properly. And this function hasn't changed a single line of code from 4.x all the way to current mainline, it's still the exact same function.

But it doesn't do any actual physical data updates ... for that reason it's value is somewhat bleak.

The only thing that will change is page vs. folio, and we can address that in the `el9` patch set.